### PR TITLE
8263041: Shenandoah: Cleanup C1 keep alive barrier check

### PR DESCRIPTION
**Pull request by zhengyu123**
Please review this cleanup. It appears that JDK-8233339 missed C1 change.

Test:
- [x] hotspot_gc_shenandoah

---------
### Progress
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue
- [x] Change must be properly reviewed

### Issue
 * [JDK-8263041](https://bugs.openjdk.java.net/browse/JDK-8263041): Shenandoah: Cleanup C1 keep alive barrier check

### Reviewers
 * [Aleksey Shipilev](https://openjdk.java.net/census#shade) (@shipilev - **Reviewer**)
 * [Roman Kennke](https://openjdk.java.net/census#rkennke) (@rkennke - **Reviewer**)

### Download
`$ git fetch https://git.openjdk.java.net/jdk pull/2834/head:pull/2834`
`$ git checkout pull/2834`

### CODE REVIEW

File: src/hotspot/share/gc/shenandoah/c1/shenandoahBarrierSetC1.cpp [hunk 1]
```diff
@@ -221,26 +221,21 @@ void ShenandoahBarrierSetC1::load_at_resolved(LIRAccess& access, LIR_Opr result)
   }
 
   // 3: apply keep-alive barrier if ShenandoahSATBBarrier is set
-  if (ShenandoahSATBBarrier) {
-    bool is_weak = (decorators & ON_WEAK_OOP_REF) != 0;
-    bool is_phantom = (decorators & ON_PHANTOM_OOP_REF) != 0;
+  if (ShenandoahSATBBarrier && ShenandoahBarrierSet::need_keep_alive_barrier(decorators, type)) {
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/c1/shenandoahBarrierSetC1.cpp:224]: I wonder if it is actually valid to check for SATB barrier here. `ShenandoahBarrierSetC2::load_at_resolved` only does `need_keep_alive_barrier` check.

**Review by shipilev — CHANGES_REQUESTED** [review]

File: src/hotspot/share/gc/shenandoah/c2/shenandoahBarrierSetC2.cpp [hunk 2]
```diff
@@ -551,7 +551,7 @@ Node* ShenandoahBarrierSetC2::load_at_resolved(C2Access& access, const Type* val
     }
   }
 
-  // 3: apply keep-alive barrier if needed
+  // 3: apply keep-alive barrier for java.lang.ref.Reference if needed
```
Comment (shipilev) [inline src/hotspot/share/gc/shenandoah/c2/shenandoahBarrierSetC2.cpp:554]: "For java.lang.ref.Reference"?

I think BarrierSetC2 comment if pretty generic, so we might want to match it.

**Review by shipilev — APPROVED** [review]
Looks good, minor nit.

**Review by rkennke — APPROVED** [review]
Looks good!

**Pull request closed without merging**